test/hotspot/jtreg/compiler/vectorapi/VectorCommutativeOperSharingTest.java

```diff
@@ -0,0 +1,139 @@
+    @IR(counts = {IRNode.ADD_VI, " 2 ", IRNode.MUL_VI, " 2 ", IRNode.MAX_VI, " 2 ",
+                  IRNode.MIN_VI, " 2 "})
+    public void testVectorIRSharing1(int index) {
+        IntVector vec1 = IntVector.fromArray(I_SPECIES, ia, index);
+        IntVector vec2 = IntVector.fromArray(I_SPECIES, ib, index);
+        vec1.lanewise(VectorOperators.ADD, vec2)
+            .lanewise(VectorOperators.ADD, vec2.lanewise(VectorOperators.ADD, vec1))
+            .intoArray(ir1, index);
+        vec1.lanewise(VectorOperators.MUL, vec2)
+            .lanewise(VectorOperators.MUL, vec2.lanewise(VectorOperators.MUL, vec1))
+            .intoArray(ir2, index);
+        vec1.lanewise(VectorOperators.MAX, vec2)
+            .lanewise(VectorOperators.MAX, vec2.lanewise(VectorOperators.MAX, vec1))
+            .intoArray(ir3, index);
+        vec1.lanewise(VectorOperators.MIN, vec2)
+            .lanewise(VectorOperators.MIN, vec2.lanewise(VectorOperators.MIN, vec1))
+            .intoArray(ir4, index);
+    }
+
+    @Run(test = "testVectorIRSharing1")
+    public void testVectorIRSharingDriver1() {
+        for (int i = 0; i < I_SPECIES.loopBound(LENGTH); i += I_SPECIES.length()) {
+            testVectorIRSharing1(i);
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.XOR_VI, " 0 ", IRNode.OR_VI, " 1 ", IRNode.AND_VI, " 1 "})
+    public void testVectorIRSharing2(int index) {
+        IntVector vec1 = IntVector.fromArray(I_SPECIES, ia, index);
+        IntVector vec2 = IntVector.fromArray(I_SPECIES, ib, index);
+        vec1.lanewise(VectorOperators.XOR, vec2)
+            .lanewise(VectorOperators.XOR, vec2.lanewise(VectorOperators.XOR, vec1))
+            .intoArray(ir1, index);
+        vec1.lanewise(VectorOperators.AND, vec2)
+            .lanewise(VectorOperators.AND, vec2.lanewise(VectorOperators.AND, vec1))
+            .intoArray(ir2, index);
+        vec1.lanewise(VectorOperators.OR, vec2)
+            .lanewise(VectorOperators.OR, vec2.lanewise(VectorOperators.OR, vec1))
+            .intoArray(ir3, index);
+    }
+
+    @Run(test = "testVectorIRSharing2")
+    public void testVectorIRSharingDriver2() {
+        for (int i = 0; i < I_SPECIES.loopBound(LENGTH); i += I_SPECIES.length()) {
+            testVectorIRSharing2(i);
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.SATURATING_ADD_VI, " 2 "}, applyIfCPUFeature = {"avx2", "true"})
+    public void testVectorIRSharing3(int index) {
+        IntVector vec1 = IntVector.fromArray(I_SPECIES, ia, index);
+        IntVector vec2 = IntVector.fromArray(I_SPECIES, ib, index);
+        vec1.lanewise(VectorOperators.SADD, vec2)
+            .lanewise(VectorOperators.SADD, vec2.lanewise(VectorOperators.SADD, vec1))
+            .intoArray(ir4, index);
+    }
+
+    @Run(test = "testVectorIRSharing3")
+    public void testVectorIRSharingDriver3() {
+        for (int i = 0; i < I_SPECIES.loopBound(LENGTH); i += I_SPECIES.length()) {
+            testVectorIRSharing3(i);
+        }
+    }
+}
```

Comment:
As mentioned above, it would be good to have all combinations of 2 ops, with inputs x and y:
```
add(x,x)  with add(x,x)
add(y,x)  with add(x,x)
add(x,y)  with add(x,x)
add(y,y)  with add(x,x)
add(x,x)  with add(y,x)
add(y,x)  with add(y,x)
add(x,y)  with add(y,x)
add(y,y)  with add(y,x)
add(x,x)  with add(x,y)
add(y,x)  with add(x,y)
add(x,y)  with add(x,y)
add(y,y)  with add(x,y)
add(x,x)  with add(y,y)
add(y,x)  with add(y,y)
add(x,y)  with add(y,y)
add(y,y)  with add(y,y)
```
At least do this for one operator. All may be a bit much to write... Templates would really be fantastic here... coming soon.

Reply:
Done, I think templatization will add value in some cases.